### PR TITLE
feat: throw exception if rtstruct references images outside of series data

### LINKED ISSUE #10: No Error Message when Loading an RT-Struct while Providing an Unreferenced Series
RT-Utils should inform the user when the ROI uses a dicom series that has not been referenced in a RT-struct when loading a previously generated rt-struct.

### PR DESCRIPTION
**Pull request by asim-shrestha**
closes #10

### CODE REVIEW

File: tests/test_rtstruct_builder.py [hunk 1]
```diff
@@ -91,20 +99,20 @@ def test_loading_valid_rt_struct(series_path):
     assert hasattr(rtstruct.ds, 'ROIContourSequence')
     assert hasattr(rtstruct.ds, 'StructureSetROISequence')
     assert hasattr(rtstruct.ds, 'RTROIObservationsSequence')
-    assert len(rtstruct.ds.ROIContourSequence) == 4
```
Comment (asim-shrestha) [inline tests/test_rtstruct_builder.py:94]: Changed the test rt struct file to only have 1 ROI within it

File: rt_utils/rtstruct_builder.py [hunk 2]
```diff
@@ -29,14 +31,47 @@ def create_from(dicom_series_path: str, rt_struct_path: str) -> RTStruct:
...
                 not hasattr(ds, 'RTROIObservationsSequence'):
             raise Exception("Please check that the existing RTStruct is valid")
+
+    @staticmethod
+    def validate_rtstruct_series_references(ds: Dataset, series_data: List[Dataset]):
+        """
+        Method to validate RTStruct only references dicom images found within the input series_data
+        """
+        # Blame DICOM for this nesting
+        for refd_frame_of_ref in ds.ReferencedFrameOfReferenceSequence:
```
Comment (asim-shrestha) [inline rt_utils/rtstruct_builder.py:57]: lmk if i can make this cleaner. Also lmk if you have better ideas for method names

**Comment by awtkns** [comment]
@carluri now is your time to become a contributor (if you review an make a change) 😉

File: rt_utils/rtstruct_builder.py [hunk 3]
```diff
@@ -29,14 +31,47 @@ def create_from(dicom_series_path: str, rt_struct_path: str) -> RTStruct:
...
+        """
+        for series in series_data:
+            if contour_image.ReferencedSOPInstanceUID == series.file_meta.MediaStorageSOPInstanceUID:
+                return
+
+        # ReferencedSOPInstanceUID is NOT available
+        raise Exception(
+            f'Loaded RTStruct references image(s) that are not contained in input series data. ' + 
+            f'Problematic image has SOP Instance Id: {contour_image.ReferencedSOPInstanceUID}'
+        )
```
Comment (awtkns) [inline rt_utils/rtstruct_builder.py:77]: Do we want to maybe throw an RTStuctException?

File: rt_utils/rtstruct_builder.py [hunk 4]
```diff
@@ -29,14 +31,47 @@ def create_from(dicom_series_path: str, rt_struct_path: str) -> RTStruct:
...
                 not hasattr(ds, 'StructureSetROISequence') or \
                 not hasattr(ds, 'RTROIObservationsSequence'):
             raise Exception("Please check that the existing RTStruct is valid")
+
+    @staticmethod
+    def validate_rtstruct_series_references(ds: Dataset, series_data: List[Dataset]):
+        """
+        Method to validate RTStruct only references dicom images found within the input series_data
+        """
+        # Blame DICOM for this nesting
```
Comment (awtkns) [inline rt_utils/rtstruct_builder.py:56]:
```suggestion

```

**Pull request merged**